File: Numbers/Fibonacci_With_Two_Variables/fib.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(fib(6) == 8)
	fmt.Println(fib(7) == 13)
	fmt.Println(fib(8) == 21)
}

func fib(n int) int {
	v1 := 0
	v2 := 1
	for i := 1; i < n; i++ {
		v2 = v1 + v2
		v1 = v2 - v1
	}
	return v2
}

File: Numbers/Fibonacci_With_Two_Variables/Fib.java
package Numbers.Fibonacci_With_Two_Variables;

public class Fib {
    public static void main(String... args) {

        System.out.println(fib(6) == 8);
        System.out.println(fib(7) == 13);
        System.out.println(fib(8) == 21);

    }

    public static int fib(int n) {
        int v1 = 0;
        int v2 = 1;
        for (int i = 1; i < n; i++) {
            v2 = v1 + v2;
            v1 = v2 - v1;
        }
        return v2;
    }
}
